src/Components/ProgramListPage.js

```diff
@@ -1,22 +1,101 @@
-import React from 'react';
-import { StyleSheet, View, FlatList } from 'react-native';
+import React, { useState } from 'react';
+import { ScrollView, StyleSheet, Text, View, TouchableOpacity, FlatList, Image } from 'react-native';
 import { connect } from 'react-redux';
 
+import * as Animatable from 'react-native-animatable';
+import Accordion from 'react-native-collapsible/Accordion';
 import theme from '../styles/theme.style';
 import Program from './programs/Program';
 import Button from './shared/Button';
+import doneImage from '../../assets/button_validation_dark_theme.png';
+import toDoImage from '../../assets/button_validation_light.png';
 
 export const ProgramListPage = props => {
-  const { programs, navigation } = props;
+  const { navigation, programs } = props;
+
+  const [activeSections, setActiveSections] = useState([undefined]);
+
+  const setSections = sections => {
+    setActiveSections(sections.includes(undefined) ? [] : sections);
+  };
+
+  const renderHeader = (section, _, isActive) => {
+    return (
+      <Animatable.View
+        duration={400}
+        style={[styles.header, isActive ? styles.active : styles.inactive]}
+        transition="backgroundColor"
+      >
+        <View>
+          <Program program={section} />
+        </View>
+      </Animatable.View>
+    );
+  };
+
+  const renderTraining = ({ item }) => {
+    const training = item;
+    const program = programs[activeSections];
+    const onTrainingPress = () => navigation.navigate('TrainingPage', { training, program });
+
+    const isDone = program
+      ? props.completeTrainings.find(
+          complete => program.id === complete.program.id && training.id === complete.training.id,
+        ) !== undefined
+      : false;
+
+    return (
+      <View>
+        <TouchableOpacity style={styles.training} onPress={onTrainingPress}>
+          <View style={styles.titleContainer}>
+            <Text style={styles.titleAccordion}>{item.title}</Text>
+          </View>
+          <View style={styles.containerAccordion}>
+            <Image style={styles.todoState} source={isDone ? doneImage : toDoImage} />
+          </View>
+        </TouchableOpacity>
+      </View>
+    );
+  };
+
+  const renderContent = (section, _, isActive) => {
```

Comment:
It would probably make it a lot more readable to extract most of this `renderContent` and `renderTraining` to another component (just as we did for `Program` last time)
Then it would be easier to understand